components/script/dom/urlhelper.rs

```diff
@@ -43,6 +43,25 @@ impl UrlHelper {
         let _ = wrapper.set_host(&value.0);
     }
 
+    pub fn Origin(url: &Url) -> USVString {
+        USVString(match url.origin() {
+            Origin::UID(_) => {
+                // https://html.spec.whatwg.org/multipage/#unicode-serialisation-of-an-origin
+                // If the origin in question is not a scheme/host/port tuple,
+                // then return the literal string "null" and abort these steps.
+                "null".to_owned()
+            },
+            Origin::Tuple(protocol, host, port) => {
+                format!(
+                    "{protocol}://{host}:{port}",
```

Comment:
good call, the port is now only appended if it's not a default port.

(as determined by the url.scheme_data.port field, which will be None if the port is a default port
https://github.com/servo/rust-url/blob/d6e1ac513e5f6999e651418fbec9d7a0ce2037e3/src/lib.rs#L274)